File: app-miyou/src/main/java/com/miyou/app/application/mission/usecase/MissionQueryUseCase.java
package com.miyou.app.application.mission.usecase;

import com.miyou.app.domain.dialogue.model.UserId;
import com.miyou.app.domain.mission.model.Mission;
import com.miyou.app.domain.mission.model.UserMission;
import reactor.core.publisher.Flux;

public interface MissionQueryUseCase {

	Flux<Mission> getAllMissions();

	Flux<UserMission> getUserMissions(UserId userId);
}

File: app-miyou/src/main/kotlin/com/miyou/app/application/mission/usecase/MissionQueryUseCase.kt
package com.miyou.app.application.mission.usecase

import com.miyou.app.domain.dialogue.model.UserId
import com.miyou.app.domain.mission.model.Mission
import com.miyou.app.domain.mission.model.UserMission
import reactor.core.publisher.Flux

interface MissionQueryUseCase {
    fun getAllMissions(): Flux<Mission>

    fun getUserMissions(userId: UserId): Flux<UserMission>
}
